internal/gengapic/gengapic.go

```diff
@@ -243,12 +243,49 @@ func (g *generator) deadline(s, m string) {
 	g.imports[pbinfo.ImportSpec{Path: "time"}] = true
 }
 
-func (g *generator) insertMetadata(m *descriptor.MethodDescriptorProto) error {
+func (g *generator) getFormattedValue(m *descriptor.MethodDescriptorProto, field string, accessor string) (string, error) {
+	f := g.lookupField(m.GetInputType(), field)
+	value := ""
+	// TODO(noahdietz): need to handle []byte for TYPE_BYTES.
+	switch f.GetType() {
+	case descriptor.FieldDescriptorProto_TYPE_STRING:
+		value = fmt.Sprintf("url.QueryEscape(%s)", accessor)
+	case descriptor.FieldDescriptorProto_TYPE_DOUBLE:
+		// Double and float are handled the same way.
+		fallthrough
+	case descriptor.FieldDescriptorProto_TYPE_FLOAT:
+		// Format the floating point value with mode 'g' to allow for
+		// exponent formatting when necessary, and decimal when adequate.
+		// QueryEscape the resulting string in case there is a '+' in the
+		// exponent.
+		// See golang.org/pkg/fmt for more information on formatting.
+		value = fmt.Sprintf(`url.QueryEscape(fmt.Sprintf("%%g", %s))`, accessor)
+	case descriptor.FieldDescriptorProto_TYPE_ENUM:
+		en := g.descInfo.Type[f.GetTypeName()]
+
+		n, imp, err := g.descInfo.NameSpec(en)
+		if err != nil {
+			return "", err
+		}
+		g.imports[imp] = true
+
+		// protobuf Go generates a mapping from number to string
+		// representation of an enum, in UPPER_SNAKE_CASE form. The map
+		// is named with the enum name and the _name suffix. If it is a
+		// nested enum, the name is prefixed with the parent message name.
+		// For example, Severity_name or Error_Severity_name.
+		value = fmt.Sprintf("%s.%s_name[int32(%s)]", imp.Name, n, accessor)
+	default:
+		value = accessor
+	}
+	return value, nil
+}
+
+func (g *generator) insertRegularRequestHeaders(m *descriptor.MethodDescriptorProto) error {
```

Comment:
Could we rename this to `insertImplicitRequestHeaders`? There is no "regular" in our world just one or the other :)

Reply:
ha - makes sense. Done!